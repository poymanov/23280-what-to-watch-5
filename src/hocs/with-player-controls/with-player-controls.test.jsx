import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withPlayerControls from "./with-player-controls";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withPlayerControls(MockComponent);

const noop = () => {};

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

it(`withPlayerControls is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped movie={movie} onChangeFullscreen={noop} onPlayerClose={noop}>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
