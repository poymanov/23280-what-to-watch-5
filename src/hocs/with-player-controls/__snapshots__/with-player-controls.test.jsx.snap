// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPlayerControls is rendered correctly 1`] = `
<div
  className="player"
>
  <div />
</div>
`;
